applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_3d.py

```diff
@@ -108,11 +108,18 @@ def __ComputeLowerSurfaceNormals(self):
         for cond in self.body_model_part.Conditions:
             # The surface normal points outisde the domain
             surface_normal = cond.GetGeometry().Normal()
+            norm = math.sqrt(surface_normal[0]**2 + surface_normal[1]**2 + surface_normal[2]**2)
+            # Normalizing normal vector
+            surface_normal /= norm
```

Comment:
This gives a `nan` if `norm` is zero. You must check it.

Reply:
You are right! I added the check :+1:

Reply:
I'd check against an epsilon rather than 0.0.